File: CSharp/SimpleOperations/PerspectiveTransform.cs
using System;
using System.Diagnostics;
using Kitware.VTK;

void PerspectiveTransform() {
   vtkMatrix4x4 m = vtkMatrix4x4.New();
   m.SetElement(0, 0, 1);
   m.SetElement(0, 1, 2);
   m.SetElement(0, 2, 3);
   m.SetElement(0, 3, 4);
   m.SetElement(1, 0, 2);
   m.SetElement(1, 1, 2);
   m.SetElement(1, 2, 3);
   m.SetElement(1, 3, 4);
   m.SetElement(2, 0, 3);
   m.SetElement(2, 1, 2);
   m.SetElement(2, 2, 3);
   m.SetElement(2, 3, 4);
   m.SetElement(3, 0, 4);
   m.SetElement(3, 1, 2);
   m.SetElement(3, 2, 3);
   m.SetElement(3, 3, 4);

   vtkPerspectiveTransform perspectiveTransform = vtkPerspectiveTransform.New();
   perspectiveTransform.SetMatrix(m);

   vtkTransform transform = vtkTransform.New();
   transform.SetMatrix(m);

   double[] p = new double[]{1.0, 2.0, 3.0};

   double[] normalProjection = transform.TransformPoint(p[0], p[1], p[2]);

   Console.WriteLine("Standard projection: "
      + normalProjection[0] + " "
      + normalProjection[1] + " "
      + normalProjection[2]);

   double[] perspectiveProjection = perspectiveTransform.TransformPoint(p[0], p[1], p[2]);

   Console.WriteLine("Perspective  projection: "
      + perspectiveProjection[0] + " "
      + perspectiveProjection[1] + " "
      + perspectiveProjection[2]);
}

File: src/Java/SimpleOperations/PerspectiveTransform.java
import vtk.vtkNativeLibrary;
import vtk.vtkPerspectiveTransform;
import vtk.vtkTransform;
import vtk.vtkMatrix4x4;

public class PerspectiveTransform 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------
	
  public static void main(String s[]) 
  {
    vtkMatrix4x4 M = new vtkMatrix4x4();
    M.SetElement(0, 0, 1);
    M.SetElement(0, 1, 2);
    M.SetElement(0, 2, 3);
    M.SetElement(0, 3, 4);
    M.SetElement(1, 0, 2);
    M.SetElement(1, 1, 2);
    M.SetElement(1, 2, 3);
    M.SetElement(1, 3, 4);
    M.SetElement(2, 0, 3);
    M.SetElement(2, 1, 2);
    M.SetElement(2, 2, 1);
    M.SetElement(2, 3, 4);
    M.SetElement(3, 0, 4);
    M.SetElement(3, 1, 2);
    M.SetElement(3, 2, 3);
    M.SetElement(3, 3, 4);

    vtkPerspectiveTransform PT = new vtkPerspectiveTransform();
    PT.SetMatrix(M);

    vtkTransform Transform = new vtkTransform();
    Transform.SetMatrix(M);

    double p[] = new double[] {1.0,2.0,3.0};

    double normalProjection[] = new double[3];
    Transform.TransformPoint(p, normalProjection);
    System.out.println(normalProjection[0] + " " + normalProjection[1] + " " + normalProjection[2]);

    double perspectiveProjection[] = new double[3];
    PT.TransformPoint(p, perspectiveProjection);
    System.out.println(perspectiveProjection[0] + " " + perspectiveProjection[1] + " " + perspectiveProjection[2]);
  }		  
}
